File: onboarding-servey/src/main/java/com/innercircle/onboardingservey/domain/QuestionOption.java
package com.innercircle.onboardingservey.domain;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Entity
@Table(name = "questions_options")
@RequiredArgsConstructor
@Getter
public class QuestionOption {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long questionOptionId;
    private final String title;
    private final Integer order;

    @ManyToOne(fetch = FetchType.LAZY)
    private Question question;
}

File: survey-domain/src/main/kotlin/com/innercircle/surveydomain/model/QuestionOption.kt
@file:Suppress("ktlint:standard:no-wildcard-imports")

package com.innercircle.surveydomain.model

import jakarta.persistence.*

@Entity
@Table(name = "question_options")
data class QuestionOption(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int = 0,
    @ManyToOne @JoinColumn(name = "question_id", nullable = false)
    val question: Question,
    @Column(nullable = false, length = 255)
    val optionText: String,
    @Column(nullable = false)
    val optionOrder: Int,
)
